### ISSUE TITLE
Make `$(Architecture)` initialization resilient against pre-set values in `$env`

### ISSUE THREAD

**Issue by vatsan-madhavan** [dotnet/wpf#2842]
See https://github.com/cmderdev/cmder/issues/2290 for details. 

> I was building [dotnet/wpf](https://github.com/dotnet/wpf) just now and noticed that it had trouble building correctly.
> 
> I debugged it down to [these lines](https://github.com/dotnet/wpf/blob/878a6b748b66c6dc464dd50753fff5e5a2e6acdb/eng/WpfArcadeSdk/tools/Wpf.Cpp.props#L19-L21) that fail to initialize the MSBuild property `$(Architecture)` correctly (it's initialized to `64` instead of `x64`), which in turn lead to my (dev) build failures in **dotnet/wpf**.
> 
> ```
>     <Architecture Condition="'$(Platform)'=='x64' or '$(Platform)'=='x86' or '$(Platform)'=='arm' or '$(Platform)'=='arm64'">$(Platform)</Architecture>
>     <Architecture Condition="'$(Platform)'=='Win32'">x86</Architecture>
>     <Architecture Condition="'$(Architecture)'==''">x64</Architecture>
> ```
> 

**cmder** sets `$env:architecture` to values like `32`, `64`, which confuses this logic. 

https://github.com/cmderdev/cmder/blob/1071221468f2f75f2b63e482deab69c9ca986219/vendor/init.bat#L125-L132
```
:: Pick right version of clink
if "%PROCESSOR_ARCHITECTURE%"=="x86" (
    set architecture=86
    set architecture_bits=32
) else (
    set architecture=64
    set architecture_bits=64
)
```

This only affects dev-builds, and only on some alternate-consoles. Still the fix is trivial and we can make this logic a bit resilient.

**Comment by miloush** [comment]
Wouldn't it be more proper to fix the offending consoles rather than normalizing non-standard practices?

**Comment by vatsan-madhavan** [comment]
Absolutely! (BTW cmder has already fixed this based on my report - https://github.com/cmderdev/cmder/pull/2294)

That said, it assumes that the offending console is doing something unequivocally wrong. There is no universal compact on the valid values for an environment variable named `architecture`, unfortunately. For e.g., a console could have set it legitimately to something like `amd64`, which was quite commonplace before Intel jumped into the game and `x64` became the new norm. and `$(Architecture)=amd64` would still trigger the same problem on our side. 

There is a subtle _bug_ here, which is this: We don't rule out invalid values that we are incapable of dealing with further down the line. I had assumed at the time I built this that `$(Architecture)` would only be supplied by the build scripts (if they are supplied by an external party at all), and that the values supplied would only be valid ones. 

What we need here is a validity check, and possibly also a check for `amd64`. 


```xml
    <!-- 
        Architecture is intended to be driven entirely by $(Platform) - clear it first before proceeding
        to infer its value again in case it has been set externally by the environment or supplied as a 
        parameter to MSBuild 
    -->
    <Architecture/>
    <Architecture Condition="'$(Platform)'=='x64' or 
                             '$(Platform)'=='x86' or 
                             '$(Platform)'=='arm' or 
                             '$(Platform)'=='arm64'">$(Platform)</Architecture>

    <Architecture Condition="'$(Platform)'=='Win32'">x86</Architecture>
    <Architecture Condition="'$(Platform)'=='amd64'">x64</Architecture>
    <Architecture Condition="'$(Architecture)'==''">x64</Architecture>

    <!-- No suitable value was found - select a default --> 
    <Architecture Condition="'$(Architecture)' !='x86' And 
                             '$(Architecture)' !='x64' And 
                             '$(Architecture)' !='arm' And 
                             '$(Architecture)' !='arm64'">x86</Architecture>
```

**note:** _In theory, we can't deal with `arm` or `arm64` either yet; I don't expect we'll ever learn to deal with `arm(32)`.._
